docs/source/daml/reference/contract-keys.rst

```diff
@@ -0,0 +1,95 @@
+.. Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Contract keys
+#############
+
+Contract keys are an optional addition to templates. They let you specify a way of uniquely identifying contract instances, using the parameters to the template - similar to a primary key for a database.
+
+You can use contract keys to stably refer to a contract, even through iterations of instances of it.
+
+Here's an example of setting up a contract key for a bank account, to act as a bank account ID:
+
+.. literalinclude:: ../code-snippets/ContractKeys.daml
+   :language: daml
+   :start-after: -- start contract key setup snippet
+   :end-before: -- end contract key setup snippet
+
+What can be a contract key
+**************************
+
+The key should either be a tuple or a record, and it **must** include every party that you specify as a ``maintainer`` (see `Specifying maintainers`_ below).
+
+For example, with ``maintainer bank``, ``key (bank, number) : (Party, Text)`` is valid; but wouldn't be if you removed ``bank`` from the key.
+
+It's best to use simple types for your keys like ``Text`` or ``Integer``, rather than a list or more complex type.
```

Comment:
this sounds like advice but it's currently an order

Reply:
you can't use a list?

Reply:
I misunderstood what you were saying here, please ignore this comment.

You cannot _form_ new lists in the key, but you can use keys that are arguments.

Reply:
```suggestion
It's best to use simple types for your keys like ``Text`` or ``Int``, rather than a list or more complex type.
```